src/mem/cache/tags/indexing_policies/IndexingPolicies.py

```diff
@@ -44,6 +44,9 @@ class BaseIndexingPolicy(SimObject):
     # Get the associativity
     assoc = Param.Int(Parent.assoc, "associativity")
 
+    # number of bits in the tag
+    tag_bits = Param.Unsigned(64, "number of bits in the tag")
```

Comment:
Can you improve the help string here and/or improve the name of the variable? There are a few confusions possible here.
- The number of bits in the tag would really be 64 - index bits - offset bits, so 64 bits for the tag is misleading.
- What would this parameter be used for? Why would you set the number of tag bits to not be *all* of the remaining bits? Adding this explanation to the help string will enable others to understand and use this feature.

Reply:
I used 64 so that I don't change the current semantics of the code. I am open to suggestions on what to use here, so please let me know.

There are cases where the tag is shortened either by hashing or dropping the upper bits that do not change often to save storage. BTB is one example which uses the reduced tags. I added a patch showing how the reduced tag is used in BTB in the series.